main.py

```diff
@@ -436,19 +446,26 @@ def logging_toggle(*args):
 
 # A object used to pass error messages from the kernel callers to the event handlers
 class errtype:
-    def __init__(self, err, argtype):
+    def __init__(self, err: Exception, argtype: str):
+
         self.err = err
-        owner = f"<@!{BOT_OWNER[0]}>" if BOT_OWNER else "BOT OWNER"
+        owner: str = f"<@!{BOT_OWNER[0]}>" if BOT_OWNER else "BOT OWNER"
         self.errmsg = f"FATAL ERROR in {argtype}\nPlease contact {owner}\nErr: `{type(err).__name__}: {err}`"
 
+        traceback.print_exception(type(self.err), self.err, self.err.__traceback__)
+
+        with open("err.log", "a+") as logfile:
+            logfile.write(f"AT {time.strftime('%a, %d %b %Y %H:%M:%S', datetime.datetime.utcnow().utctimetuple())}:\n")
+            logfile.write("".join(traceback.format_exception(type(self.err), self.err, self.err.__traceback__)))
 
-# Catch errors.
+
+# Catch errors
 @Client.event
 async def on_error(event, *args, **kwargs):
     raise
```

Comment:
I am 90% sure it does this on its own

Reply:
well removing it makes it crash on error so clearly not